src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Mapping/RazorEditHelperTest.cs

```diff
@@ -0,0 +1,800 @@
+            }
+        }
+        """,
+        expectedRazorSource:
+        """
+        @code {
+            public int NewCounter { get; set; } 
+        }
+        """);
+
+    [Fact]
+    public Task NewUsing_TopOfFile()
+        => TestAsync(
+        csharpSource:
+        """
+        class MyComponent : ComponentBase
+        {
+            {|map1:public int Counter { get; set; }|}
+        }
+        """,
+        razorSource:
+        """
+        @code {
+            {|map1:public int Counter { get; set; }|} 
+        }
+        """,
+        newCSharpSource:
+        """
+        using System;
+
+        class MyComponent : ComponentBase
+        {
+            public int NewCounter { get; set; }
+        }
+        """,
+        expectedRazorSource:
+        """
+        @using System
+        @code {
+            public int NewCounter { get; set; } 
+        }
+        """);
+
+    [Fact]
+    public Task RemovedUsing_IsRemoved()
+        => TestAsync(
+        csharpSource:
+        """
+        {|mapUsing:using System;|}
+
+        class MyComponent : ComponentBase
+        {
+            {|map1:public int Counter { get; set; }|}
+        }
+        """,
+        razorSource:
+        """
+        {|mapUsing:@using System|}
+
+        @code {
+            {|map1:public int Counter { get; set; }|} 
+        }
+        """,
+        newCSharpSource:
+        """
+        class MyComponent : ComponentBase
+        {
+            public int Counter { get; set; }
+        }
+        """,
+        expectedRazorSource:
+        """
+
+        @code {
+            public int Counter { get; set; } 
+        }
+        """);
+
+    [Fact]
+    public Task NewUsing_AfterExisting()
+        => TestAsync(
+        csharpSource:
+        """
+        {|mapUsing:using System;|}
+
+        class MyComponent : ComponentBase
+        {
+            {|map1:public int Counter { get; set; }|}
+        }
+        """,
+        razorSource:
+        """
+        {|mapUsing:@using System|}
+
+        @code {
+            {|map1:public int Counter { get; set; }|} 
+        }
+        """,
+        newCSharpSource:
+        """
+        using System;
+        using System.Collections.Generic;
+
+        class MyComponent : ComponentBase
+        {
+            public int NewCounter { get; set; }
+        }
+        """,
+        expectedRazorSource:
+        """
+        @using System
+        @using System.Collections.Generic
+
+        @code {
+            public int NewCounter { get; set; } 
+        }
+        """);
+
+    [Fact]
+    public Task NewUsing_AfterPage()
+        => TestAsync(
+        csharpSource:
+        """
+        class MyComponent : ComponentBase
+        {
+            {|map1:public int Counter { get; set; }|}
+        }
+        """,
+        razorSource:
+        """
+        @page "/counter"
+
+        @code {
+            {|map1:public int Counter { get; set; }|} 
+        }
+        """,
+        newCSharpSource:
```

Comment:
Okay this is the pettiest of nits, but perhaps it would be easier to rearrange the parameters so its C#, new C#, Razor, new Razor. Would make it easier to tell what has changed between new and old I think